### ISSUE TITLE
Android: not logging user out after inactivity

### ISSUE THREAD

**Issue by mark-prins** [msupply-foundation/open-msupply#3967]
## What went wrong? 😲

Issue raised by Nisha via telegram, validated by Maria & Adam, as noted:

**Nisha**
If we just leave the tablet with omsupply open then this kind of wrong alert arises in every page which is Authentication error permission denied. 

https://github.com/msupply-foundation/open-msupply/assets/9192912/8c17ca68-fe8a-4540-9964-251d564cddac

**Maria**
I noticed that too! Sometimes when I leave it open on the synchronisation screen for too long, that message pops up but the sync is still happening in the background. The name of the store and the admin area disappears though

![telegram-cloud-photo-size-2-5345975398997024008-y](https://github.com/msupply-foundation/open-msupply/assets/9192912/439745b3-3b15-4050-bc9e-6c5f2e862edd)

![telegram-cloud-photo-size-2-5345975398997024009-y](https://github.com/msupply-foundation/open-msupply/assets/9192912/9fb88304-fca9-4cbd-a866-c9026fa8237e)

**Adam**
I've definitely noticed that the tablet doesn't actually log you out when the timeout is reached, so it lets you do some actions and then suddenly says "oh, actually you're not logged in" - it's a bit of a frustrating UX to be honest (and possibly negates the security reason for a timeout if you can still access some info if the screen is left on?)

**Nisha**
this one, I’ve also seen many times…but the steps are not really traceable. 
At times, it logs me out after few mins of inactivity, while sometimes, the session is still active even after leaving the screen inactive for more than 30 mins. 🤡

**Maria**
I’ve only experienced the timeout when leaving the tablet on & inactive in the admin section - has it been happening in other areas of omSupply?

**Adam**
Yeah, it's happened in random places for me - I last touched this over 30 mins ago on this screen
So will probably let me do a little stuff if I start playing with it and then I'll run into an issue and it will log me out

## Expected behaviour 🤔
I think there should be 'You're not logged in....' message instead of 'Permission denied' message in that alert

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Log in to the android app
2. Leave alone for an hour+

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version: Android

- Platform:

- Database type: SQLite3

**Comment by andreievg** [comment]
daily triage: Setting as normal because the app is still quite usable, but does need improvement.

I remember at one point making a diagram of auth logic and hooks on front end, and after about 1 hour the diagram started to get really messy and i was getting confused about the flow. I feel like for a permanent fix for this we may need to do some refactoring, or at the very least a detail diagram of how all of the actions/events interact with each other.

**Comment by mark-prins** [comment]
removing from 2.2 due to normal priority

**Comment by CarlosNZ** [comment]
> I feel like for a permanent fix for this we may need to do some refactoring, or at the very least a detail diagram of how all of the actions/events interact with each other.

Yeah, I'd be keen to have a crack at this when it becomes higher priority.

**Comment by jmbrunskill** [comment]
https://github.com/msupply-foundation/open-msupply/issues/694 related?

**Comment by DhanyaHerath** [comment]
Triage: Also test #6943 when fixing
